File: BusinessLogic/Player.cs
using System;

namespace CocaineCartels.BusinessLogic
{
    public class Player
    {
        internal Player(Guid id, string colors, string name)
        {
            Id = id;
            Color = colors;
            Name = name;

            CommandsSentOn = null;
            Points = 0;
            PointsLastTurn = 0;
            Ready = false;
        }

        internal Guid Id { get; }

        /// <summary>The main color also identifies a player, so use this property the player ID.</summary>
        public string Color { get; }

        public DateTime? CommandsSentOn { get; set; }

        public string Name { get; }

        public int Points { get; set; }

        public int PointsLastTurn { get; set; }

        /// <summary>If the game hasn't started yet: The player believe that all players are here. If the game has started: The player has sent in commands and is thus ready for the next turn.</summary>
        public bool Ready { get; set; }
    }
}

File: WebApplication/Client/Player.ts
module CocaineCartels {
    "use strict";

    export class Player {
        constructor(
            playerData: IPlayer
        ) {
            this.color = playerData.color;
            this.commandsSentOn = Player.parseDateString(playerData.commandsSentOn);
            this.points = playerData.points;
            this.pointsLastTurn = playerData.pointsLastTurn;
            this.name = playerData.name;
            this.ready = playerData.ready;
            this.sortValue = playerData.sortValue;
        }

        public color: string;
        public commandsSentOn: Date;
        public name: string;
        public points: number;
        public pointsLastTurn: number;
        public ready: boolean;
        public sortValue: number;

        /** Returns the number of move commands that the current has assigned. */
        public get numberOfMoveCommands(): number {
            const numberOfMoveCommands = Main.game.currentTurn.moveCommands.filter(command => command.player.color === this.color).length;
            return numberOfMoveCommands;
        }

        private static parseDateString(dateString: string): Date {
            if (dateString == null) {
                return null;
            }

            return new Date(dateString);
        }
    }
}
